### PR TITLE
feature/rename

### PR DESCRIPTION
**Pull request by jeroenwienk**
To test this in the web app you can set app version to >5.0.0.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -12,13 +12,16 @@ const { HomeyAPI } = require('athom-api');
 const fetch = require('node-fetch');
 const _ = require('lodash');
 
+const { uuid } = require('./lib/uuid')
```
Comment (WeeJeWel) [inline app.js:15]: Prefer `npm i uuid`

File: app.js [hunk 2]
```diff
@@ -12,13 +12,16 @@ const { HomeyAPI } = require('athom-api');
...
+
 module.exports = class HomeyScriptApp extends Homey.App {
 
   static RUN_TIMEOUT = 1000 * 30; // 30s
 
   async onInit() {
     // Init Scripts
     this.scripts = this.homey.settings.get('scripts');
+    this.migrations = this.homey.settings.get('migrations') || [];
```
Comment (WeeJeWel) [inline app.js:24]: No need for this. Simply update all scripts where `name === undefined`.

File: app.js [hunk 3]
```diff
@@ -325,16 +371,45 @@ module.exports = class HomeyScriptApp extends Homey.App {
...
+
+    return newScript;
+  }
+
+  async updateScript({ id, name, code, lastExecuted }) {
+    this.scripts[id] = {
+      ...this.scripts[id],
+    };
+
+    if (name != null) {
```
Comment (WeeJeWel) [inline app.js:393]: Explicitly check for a string. Name might be `undefined`.

File: app.js [hunk 4]
```diff
@@ -325,16 +371,45 @@ module.exports = class HomeyScriptApp extends Homey.App {
...
+      this.scripts[id].name = name;
+    }
+
+    if (code != null) {
```
Comment (WeeJeWel) [inline app.js:397]: Explicitly check for a string. Code might be undefined.

File: app.js [hunk 5]
```diff
@@ -325,16 +371,45 @@ module.exports = class HomeyScriptApp extends Homey.App {
...
+      this.scripts[id].code = code;
+    }
+
+    if (lastExecuted != null) {
```
Comment (WeeJeWel) [inline app.js:401]: Explicitly check for a Date. lastExecuted might be undefined.

**Review by WeeJeWel — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (jeroenwienk) [reply app.js]: Done

File: app.js [hunk 6]
```diff
@@ -74,45 +75,88 @@ module.exports = class HomeyScriptApp extends Homey.App {
       this.homey.settings.set('scripts', this.scripts);
     }
 
+    if (this.scripts) {
```
Comment (WeeJeWel) [inline app.js:78]:
```suggestion
// Migration: Add missing `name` property
if (this.scripts) {
  let anyScriptChanged = false;
  
  for( const script of Object.values(this.scripts) ) {
    if( typeof script.name !== 'string' ) {
      script.name = script.id;
      anyScriptChanged = true;
    }
  }
  
  if (anyScriptChanged) {
    this.homey.settings.set('scripts', this.scripts);
  }
}
```

**Review by WeeJeWel — APPROVED** [review]
See suggestion.

File: app.js (hunk 6 above)
Comment (jeroenwienk) [reply app.js:78]: Scripts dont have an id property

**Pull request merged**